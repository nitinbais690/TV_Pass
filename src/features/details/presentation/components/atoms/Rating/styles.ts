import { StyleSheet } from 'react-native';
import { appFonts, appPadding } from 'core/styles/AppStyles';
import { scale, selectDeviceType } from 'qp-common-ui';

export const styles = (appColors: any) => {
    const paddingHorizontal = selectDeviceType({ Handset: scale(4) }, scale(7));
    return StyleSheet.create({
        container: {
            flexDirection: 'row',
            height: scale(18),
            backgroundColor: appColors.primaryVariant7,
            borderColor: '#35393F',
            borderRadius: scale(5),
            borderStyle: 'solid',
            borderWidth: scale(1),
            shadowColor: appColors.shadow,
            shadowOffset: { width: 0, height: scale(2.7) },
            shadowRadius: scale(5),
            elevation: scale(2.7),
            paddingHorizontal: paddingHorizontal,
            justifyContent: 'center',
            alignItems: 'center',
            marginEnd: appPadding.xs(),
        },
        textStyle: {
            color: appColors.secondary,
            fontFamily: appFonts.primary,
            fontSize: appFonts.xxxs,
            letterSpacing: 0.13,
            fontWeight: '600',
        },
        image: {
            marginEnd: paddingHorizontal,
        },
    });
};
