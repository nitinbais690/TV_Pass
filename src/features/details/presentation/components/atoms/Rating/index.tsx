import React from 'react';
import { View, Text } from 'react-native';
import { styles } from './styles';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import { useLocalization } from 'contexts/LocalizationContext';
import PremiumTick from 'assets/images/premium_tick.svg';

export default function Rating(props: { rating: string; icon?: string }) {
    const { strings } = useLocalization();
    let ratingStyle = styles(useAppColors());

    const checkisPremiumTag = (): boolean => {
        return props.rating === strings.premium;
    };

    return (
        <View style={ratingStyle.container}>
            {checkisPremiumTag() && <PremiumTick style={ratingStyle.image} />}
            <Text style={ratingStyle.textStyle}>{props.rating}</Text>
        </View>
    );
}
