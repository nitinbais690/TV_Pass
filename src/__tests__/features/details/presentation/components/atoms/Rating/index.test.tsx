import React from 'React';
import { render, toJSON } from '@testing-library/react-native';
import Rating from 'features/details/presentation/components/atoms/Rating';

jest.mock('core/presentation/hooks/use-app-colors');

describe('rating', () => {
    test('renders correctly', () => {
        const { container } = render(<Rating rating={'Violence'} icon={'assets/images/vln.png'} />);
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render without icon', () => {
        const { container } = render(<Rating rating={'Violence'} />);
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with empty text', () => {
        const { container } = render(<Rating rating={''} />);
        expect(toJSON(container)).toMatchSnapshot();
    });
});
