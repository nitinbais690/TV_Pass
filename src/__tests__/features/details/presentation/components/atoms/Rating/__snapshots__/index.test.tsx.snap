// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rating render with empty text 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1A1D1E",
        "borderColor": "#35393F",
        "borderRadius": 7.285714285714286,
        "borderStyle": "solid",
        "borderWidth": 1.457142857142857,
        "elevation": 3.934285714285714,
        "flexDirection": "row",
        "height": 26.228571428571428,
        "justifyContent": "center",
        "marginEnd": "2%",
        "paddingHorizontal": 10.2,
        "shadowColor": "#40000000",
        "shadowOffset": Object {
          "height": 3.934285714285714,
          "width": 0,
        },
        "shadowRadius": 7.285714285714286,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 12,
          "fontWeight": "600",
          "letterSpacing": 0.13,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`rating render without icon 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1A1D1E",
        "borderColor": "#35393F",
        "borderRadius": 7.285714285714286,
        "borderStyle": "solid",
        "borderWidth": 1.457142857142857,
        "elevation": 3.934285714285714,
        "flexDirection": "row",
        "height": 26.228571428571428,
        "justifyContent": "center",
        "marginEnd": "2%",
        "paddingHorizontal": 10.2,
        "shadowColor": "#40000000",
        "shadowOffset": Object {
          "height": 3.934285714285714,
          "width": 0,
        },
        "shadowRadius": 7.285714285714286,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 12,
          "fontWeight": "600",
          "letterSpacing": 0.13,
        }
      }
    >
      Violence
    </Text>
  </View>
</View>
`;

exports[`rating renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1A1D1E",
        "borderColor": "#35393F",
        "borderRadius": 7.285714285714286,
        "borderStyle": "solid",
        "borderWidth": 1.457142857142857,
        "elevation": 3.934285714285714,
        "flexDirection": "row",
        "height": 26.228571428571428,
        "justifyContent": "center",
        "marginEnd": "2%",
        "paddingHorizontal": 10.2,
        "shadowColor": "#40000000",
        "shadowOffset": Object {
          "height": 3.934285714285714,
          "width": 0,
        },
        "shadowRadius": 7.285714285714286,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 12,
          "fontWeight": "600",
          "letterSpacing": 0.13,
        }
      }
    >
      Violence
    </Text>
  </View>
</View>
`;
